nautobot/docs/development/apps/migration/ui-component-framework/v3-ui-migration-guide.md

```diff
@@ -0,0 +1,245 @@
+# Nautobot V3 UI Migration Guide
+
+With the release of Nautobot v3, the UI has undergone significant improvements to enhance usability, flexibility, and maintainability. This version introduces an upgrade from Bootstrap 3.4.1 to Bootstrap 5. This transition brings modern UI enhancements, improved accessibility, and better responsiveness. While these changes provide a more consistent and intuitive user experience, they also introduce necessary modifications that may impact existing custom UI components and templates.
+
+The following is a detailed migration guide outlining the steps to help app authors update their app's UI for compatibility with the Bootstrap upgrade. Additional changes that are needed and might not be captured in this migration guide can be found in the Bootstrap V5 documentation.
+
+## Breadcrumb Items
+
+- Breadcrumb Item is changed from `<li>` to `<li class="breadcrumb-item">`
+- Active breadcrumb item is changed from `<li class="active">Data</li>` to `<li class="breadcrumb-item active" aria-current="page">Data</li>`
+
+Bootstrap 3:
+
+```html
+<ol class="breadcrumb">
+  <li><a href="#">Home</a></li>
+  <li><a href="#">Library</a></li>
+  <li class="active"><a href="#">Data</a></li>
+</ol>
+```
+
+Bootstrap 5:
+
+```html
+<ol class="breadcrumb">
+    <li class="breadcrumb-item"><a href="#">Home</a></li>
+    <li class="breadcrumb-item"><a href="#">Library</a></li>
+    <li class="breadcrumb-item active" aria-current="page"><a href="#">Data</a></li>
+</ol>
+```
+
+## Buttons
+
+- `.btn-default` is replaced by `.btn-secondary`
+- `.close` is replaced by `.btn-close`
+- `.btn-xs` (extra small) is removed; use `.btn-sm` as the smallest size
+
+Bootstrap 3:
+
+```html
+<button class="btn-default"></button>
+<button class="close" data-dismiss="alert">×</button>
+```
+
+Bootstrap 5:
+
+```html
+<button class="btn-secondary"></button>
+<button class="btn-close" data-bs-dismiss="alert"></button>
+```
+
+## Hide/Show Elements
+
+- `.hidden` is replaced by `d-none`
+- `.visible-*` is replaced by `d-block` or `d-inline`
+- `.hidden-xs` is replaced by `d-block d-sm-none` (hide on extra small)
+- `.visible-xs` is replaced by `d-none d-sm-block` (show on extra small)
+- `hidden` is replaced by `visually-hidden` (visually hide elements but keep them accessible to assistive technologies e.g. screen readers)
```

Comment:
Do we have ready examples of any of these?

Reply:
We do have `hidden` to `d-none` and `hidden` to `visually-hidden` in core. I included `visible-*` classes and `hidden-* `for completness sake.